docs/proposals/registry/index-server-REST-API.md

```diff
@@ -0,0 +1,243 @@
+      "Java",
+      "Quarkus"
+    ],
+    "projectType": "quarkus",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-quarkus:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "java-springboot",
+    "displayName": "Spring Boot®",
+    "description": "Spring Boot® using Java",
+    "tags": [
+      "Java",
+      "Spring"
+    ],
+    "projectType": "spring",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-springboot:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "java-vertx",
+    "displayName": "Vert.x Java",
+    "description": "Upstream Vert.x using Java",
+    "tags": [
+      "Java",
+      "Vert.x"
+    ],
+    "projectType": "vertx",
+    "language": "java",
+    "links": {
+      "self": "devfile-catalog/java-vertx:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "nodejs",
+    "displayName": "NodeJS Runtime",
+    "description": "Stack with NodeJS 12",
+    "tags": [
+      "NodeJS",
+      "Express",
+      "ubi8"
+    ],
+    "projectType": "nodejs",
+    "language": "nodejs",
+    "links": {
+      "self": "devfile-catalog/nodejs:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "python",
+    "displayName": "Python",
+    "description": "Python Stack with Python 3.7",
+    "tags": [
+      "Python",
+      "pip"
+    ],
+    "projectType": "python",
+    "language": "python",
+    "links": {
+      "self": "devfile-catalog/python:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  },
+  {
+    "name": "python-django",
+    "displayName": "Django",
+    "description": "Python3.7 with Django",
+    "tags": [
+      "Python",
+      "pip",
+      "Django"
+    ],
+    "projectType": "django",
+    "language": "python",
+    "links": {
+      "self": "devfile-catalog/python-django:latest"
+    },
+    "resources": [
+      "devfile.yaml",
+      "meta.yaml"
+    ]
+  }
+]
+```
+
+## Gets registry stack devfile
+Gets the specific registry stack devfile content from HTTP response
+
+Note: this REST API only returns the content of `devfile.yaml`, it won't return other resources in the stack
+### HTTP request
+```
+GET http://{registry host}/devfiles/{stack}
+```
+### Request parameters
+| Parameter | Description |
+| ----------| ----------- |
+| registry host | the URL/ingress that exposes registry service |
+| stack | registry stack name |
+### Request body
+The request body must be empty.
+### Request example
+```
+curl http://devfile-registry.192.168.1.1.nip.io/devfiles/nodejs
+```
+### Response example
+```yaml
+schemaVersion: 2.0.0
+metadata:
+  name: nodejs
+  version: 1.0.0
+  alpha.build-dockerfile: "https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/build/Dockerfile"
```

Comment:
This should be probably part of the spec instead of the metadata.

Reply:
it does not seem to be allowed in the devfile metadata  all https://github.com/devfile/api/blob/071eecafa93c1f6455f5b6a8c34e57ebbdb6d27a/pkg/devfile/header.go#L20
So, it may be part of devfile or component attributes, but I don't have any idea about this kind of attributes

Reply:
These fields are directly copies from the registry repo: https://github.com/devfile/registry/blob/master/stacks/nodejs/devfile.yaml#L5-L6, seems like the [devfile API reference](https://docs.devfile.io/devfile/user-guide/api-reference.html) doesn't have those fields under metadata field, I am not sure why those fields exist in the registry repo. @johnmcollier @elsony I think we can remove those fields from registry repo if they are not necessary under metadata.

Reply:
That is not part of the formal devfile spec. The node.js stack (originally copied from the odo specific repo) contains some experimental flags that are odo specific for some temporary implementation of the odo deploy function. However, those are supposed to be formally defined and replaced by the devfile 2.2. Feel free to remove the alpha.* attributes from our repo.

Reply:
Thanks @elsony for confirming, I will remove them from our registry repo.

Reply:
PR https://github.com/devfile/registry/pull/2 to remove them from our registry repo.